ikarus/finiteelements/mechanics/linearelastic.hh

```diff
@@ -206,28 +205,29 @@ public:
   }
 
   /**
-   * @brief Calculates results at a specific local position.
+   * @brief Calculates a requested result at a specific local position.
    *
-   * @param req The ResultRequirementsType object specifying the requested results.
+   * @param req The FERequirementType object holding the global solution.
    * @param local Local position vector.
-   * @param result Map to store the calculated results.
+   * @return calculated result
+   *
+   * @tparam resType The type representing the requested result.
    */
-  void calculateAt(const ResultRequirementsType& req, const Dune::FieldVector<double, Traits::mydim>& local,
-                   ResultTypeMap<double>& result) const {
+  template <ResultType resType>
+  auto calculateAt(const FERequirementType& req, const Dune::FieldVector<double, Traits::mydim>& local) const {
     using namespace Dune::Indices;
     using namespace Dune::DerivativeDirections;
     using namespace Dune;
 
-    const auto eps = strainFunction(req.getFERequirements());
-    const auto C   = materialTangent();
-    auto epsVoigt  = eps.evaluate(local, on(gridElement));
+    static_assert(resType == ResultType::linearStress, "The requested result type is NOT implemented.");
```

Comment:
The only thing I am not exactly sure is if I should keep the static assert and then still hafe the if constexpr (which only really makes sense if we have more than one possbile resultType

Reply:
I can also imagine that we throw at runtime not implemented for all cases that are not defined. Im also not sure what is better from a user perspective

Reply:
I kinda like the `static_assert`. We can leave it for now, but we may want to reassess when we have elements with more than one result.